lib/surface/compiler/eex_engine.ex

```diff
@@ -219,16 +219,41 @@ defmodule Surface.Compiler.EExEngine do
 
     dynamic_props_expr = handle_dynamic_props(dynamic_props)
 
-    {case_expression, slot_meta, slot_props} =
-      collect_slot_meta(component, templates, buffer, state)
+    if module.__use_context__? do
+      Module.put_attribute(meta.caller.module, :use_context?, true)
+    end
+
+    context_expr =
+      cond do
+        module.__slots__() == [] and not module.__use_context__? ->
+          quote generated: true do
+            %{}
+          end
+
+        is_child_component?(state) ->
+          quote generated: true do
+            Map.merge(@__context__ || %{}, the_context)
+          end
+
+        true ->
+          quote generated: true do
+            @__context__ || %{}
+          end
+      end
+
+    {gets, props_expr} = Keyword.pop(props_expr, :get, [])
```

Comment:
would this effect any component with a prop named `get`?

Reply:
Good catch! I'll restrict to `Context`.